### DISCUSSION TITLE
Error: Can't use 'tar -xzf' extract archive file... return code: 2

### DISCUSSION THREAD

**Discussion by choonkeat** [community/community#26321] [Actions]
<p><a href="https://www.githubstatus.com" rel="noopener nofollow ugc">https://www.githubstatus.com</a> says nothing is wrong. But my builds suddenly failing at</p>
<pre><code class="lang-auto">Prepare all required actions
Getting action download info
Download action repository 'actions/checkout@v2' (SHA:&lt;redacted1&gt;)
##[debug]Download 'https://api.github.com/repos/actions/checkout/tarball/&lt;redacted1&gt;' to '/home/runner/work/_actions/_temp_&lt;redacted2&gt;/&lt;redacted3&gt;.tar.gz'
##[debug]Unwrap 'actions-checkout-7884fca' to '/home/runner/work/_actions/actions/checkout/v2'
##[debug]Archive '/home/runner/work/_actions/_temp_&lt;redacted2&gt;/&lt;redacted3&gt;.tar.gz' has been unzipped into '/home/runner/work/_actions/actions/checkout/v2'.
Download action repository 'actions/cache@v2' (SHA:&lt;redacted4&gt;)
##[debug]Download 'https://api.github.com/repos/actions/cache/tarball/&lt;redacted4&gt;' to '/home/runner/work/_actions/_temp_&lt;redacted5&gt;/&lt;redacted6&gt;.tar.gz'
Error: Can't use 'tar -xzf' extract archive file: /home/runner/work/_actions/_temp_&lt;redacted5&gt;/&lt;redacted6&gt;.tar.gz. return code: 2.
##[debug]System.NotSupportedException: Can't use 'tar -xzf' extract archive file: /home/runner/work/_actions/_temp_&lt;redacted5&gt;/&lt;redacted6&gt;.tar.gz. return code: 2.
##[debug]   at GitHub.Runner.Worker.ActionManager.DownloadRepositoryActionAsync(IExecutionContext executionContext, ActionDownloadInfo downloadInfo, String destDirectory)
##[debug]   at GitHub.Runner.Worker.ActionManager.DownloadRepositoryActionAsync(IExecutionContext executionContext, ActionDownloadInfo downloadInfo)
##[debug]   at GitHub.Runner.Worker.ActionManager.PrepareActionsRecursiveAsync(IExecutionContext executionContext, PrepareActionsState state, IEnumerable`1 actions, Int32 depth, Guid parentStepId)
##[debug]   at GitHub.Runner.Worker.ActionManager.PrepareActionsAsync(IExecutionContext executionContext, IEnumerable`1 steps, Guid rootStepId)
##[debug]   at GitHub.Runner.Worker.JobExtension.InitializeJob(IExecutionContext jobContext, AgentJobRequestMessage message)
##[debug]Finishing: Set up job
</code></pre>
<p>How can I overcome <code>actions/cache</code> errors?</p>

**Comment by Yurunsoft** [comment]
<p>me too<br>
<a href="https://github.com/imiphp/imi/runs/6704164514" rel="noopener nofollow ugc">https://github.com/imiphp/imi/runs/6704164514</a></p>

**Comment by choonkeat** [comment]
<p>my fix</p>
<pre><code class="lang-diff">-        uses: actions/cache@v2
+        uses: actions/cache@v3.0.3
</code></pre>

**Comment by piotrekkr** [comment]
<p>issue here: <a href="https://github.com/actions/cache/issues/811" class="inline-onebox" rel="noopener nofollow ugc">Cache fails with zstd error - zstd -d: Cannot exec: No such file or directory · Issue #811 · actions/cache · GitHub</a></p>

**Comment by vavuthu** [comment]
<p>raised issue : <a href="https://github.com/actions/runner/issues/1923" class="inline-onebox" rel="noopener nofollow ugc">Error: Can't use 'tar -xzf' extract archive file · Issue #1923 · actions/runner · GitHub</a></p>

**Comment by ozturkberkay** [comment]
<p>We are having the same issue as well:</p>
<pre><code class="lang-auto">Current runner version: '2.291.1'
Operating System
  Ubuntu
  20.04.4
  LTS
Virtual Environment
  Environment: ubuntu-20.04
  Version: 20220529.1
  Included Software: https://github.com/actions/virtual-environments/blob/ubuntu20/20220529.1/images/linux/Ubuntu2004-Readme.md
  Image Release: https://github.com/actions/virtual-environments/releases/tag/ubuntu20%2F20220529.1
Virtual Environment Provisioner
  1.0.0.0-main-20220516-1
GITHUB_TOKEN Permissions
  Actions: write
  Checks: write
  Contents: write
  Deployments: write
  Discussions: write
  Issues: write
  Metadata: read
  Packages: write
  Pages: write
  PullRequests: write
  RepositoryProjects: write
  SecurityEvents: write
  Statuses: write
Secret source: Actions
Prepare workflow directory
Prepare all required actions
Getting action download info
Download action repository 'actions/checkout@v2' (SHA:7884fcad6b5d53d10323aee724dc68d8b9096a2e)
Download action repository 'actions/setup-python@v3' (SHA:98f2ad02fd48d057ee3b4d4f66525b231c3e52b6)
Download action repository 'actions/cache@v2' (SHA:937d24475381cd9c75ae6db12cb4e79714b926ed)
Error: Can't use 'tar -xzf' extract archive file: /home/runner/work/_actions/_temp_35e59afd-0d56-42b7-a792-72eefd477445/7c9444a5-2d98-4789-8264-cd30c5fbf484.tar.gz. return code: 2.
</code></pre>

**Comment by piotrekkr** [comment, marked as answer]
<p>Issue seems to be resolved. There were also some local cache on runners or something but it should work even with v2 now. More info in issue above.</p>

**Comment by HansBug** [comment]
<p>Me too: <a href="https://github.community/t/error-on-github-actions-job-setting-up/254075" class="inline-onebox">Error on Github Action's job setting up</a> .</p>

**Comment by joristork** [comment]
Just hit this issue. Not an issue with our codebase.
Can't use 'tar -xzf' extract archive file: /home/runner/work/_actions/_temp_789369c1-1dbf-46ab-9c4d-5b97f1e700eb/ba1fdd03-73d3-4898-a44c-ceedda73824a.tar.gz. return code: 2.

**Reply by RobPasMue** [reply]
Also experiencing it in other codebases. Seems to be an issue on GH runners...

**Reply by cgreggtcd** [reply]
I'm getting the same error on only one version of Python out of the three I'm running... 

Error: Can't use 'tar -xzf' extract archive file: /home/runner/work/_actions/_temp_5c7609bc-a3ba-4675-aa33-cfc4a69c233b/b9736f1b-28b3-4d8f-a317-c2b33bcc49db.tar.gz. return code: 2.

**Reply by raulbrito** [reply]
We're running into this issue also, in several repos.

**Reply by lovelydinosaur** [reply]
Same here, for example... https://github.com/encode/httpx/actions/runs/3196471691/jobs/5218449159

**Reply by NickGraham101** [reply]
Seeing this issue as well, though other jobs in the same workflow succeed when trying to download the same action

**Reply by RobPasMue** [reply]
It depends on the runner you are assigned I think. Even rerunning if failed sometimes works. But the problem is there.

**Reply by grelland** [reply]
Experiencing the same issue.

**Reply by DragosRomaniuc** [reply]
Same for me.

**Reply by DragosRomaniuc** [reply]
@github-staff

**Reply by sumanthkumarc** [reply]
+1 we are facing similar issue with github self hosted runner

**Reply by mdiin** [reply]
Seeing this issue as well.

**Comment by RobPasMue** [comment]
https://github.com/actions/runner/issues/1923#issuecomment-1269759795

Seems like other people are experiencing this issue again

**Reply by RobPasMue** [reply]
@github-staff could you look into this? It seems that the cache is giving problems again

**Reply by MaxJPRey** [reply]
+1 can confirm.

**Comment by p0lygun** [comment]
+1, can confirm 
<https://github.com/battlefield-portal-community/bfportal.gg/actions/runs/3196541802/jobs/5218624844>

**Reply by spicy-sauce** [reply]
Same here, please advise.

**Comment by RobPasMue** [comment]
Looks like investigation is ongoing

https://www.githubstatus.com/

![image](https://user-images.githubusercontent.com/37798125/194295402-cb0442f0-4fa3-431c-bfd3-20f57344d568.png)

**Reply by snazy** [reply]
`We are investigating reports of degraded performance for Actions.`
"Degraded performance" :rofl:

**Comment by romansainsburys** [comment]
> https://www.githubstatus.com says nothing is wrong. But my builds suddenly failing at
> 
> ```
> Prepare all required actions
> Getting action download info
> Download action repository 'actions/checkout@v2' (SHA:<redacted1>)
> ##[debug]Download 'https://api.github.com/repos/actions/checkout/tarball/<redacted1>' to '/home/runner/work/_actions/_temp_<redacted2>/<redacted3>.tar.gz'
> ##[debug]Unwrap 'actions-checkout-7884fca' to '/home/runner/work/_actions/actions/checkout/v2'
> ##[debug]Archive '/home/runner/work/_actions/_temp_<redacted2>/<redacted3>.tar.gz' has been unzipped into '/home/runner/work/_actions/actions/checkout/v2'.
> Download action repository 'actions/cache@v2' (SHA:<redacted4>)
> ##[debug]Download 'https://api.github.com/repos/actions/cache/tarball/<redacted4>' to '/home/runner/work/_actions/_temp_<redacted5>/<redacted6>.tar.gz'
> Error: Can't use 'tar -xzf' extract archive file: /home/runner/work/_actions/_temp_<redacted5>/<redacted6>.tar.gz. return code: 2.
> ##[debug]System.NotSupportedException: Can't use 'tar -xzf' extract archive file: /home/runner/work/_actions/_temp_<redacted5>/<redacted6>.tar.gz. return code: 2.
> ##[debug]   at GitHub.Runner.Worker.ActionManager.DownloadRepositoryActionAsync(IExecutionContext executionContext, ActionDownloadInfo downloadInfo, String destDirectory)
> ##[debug]   at GitHub.Runner.Worker.ActionManager.DownloadRepositoryActionAsync(IExecutionContext executionContext, ActionDownloadInfo downloadInfo)
> ##[debug]   at GitHub.Runner.Worker.ActionManager.PrepareActionsRecursiveAsync(IExecutionContext executionContext, PrepareActionsState state, IEnumerable`1 actions, Int32 depth, Guid parentStepId)
> ##[debug]   at GitHub.Runner.Worker.ActionManager.PrepareActionsAsync(IExecutionContext executionContext, IEnumerable`1 steps, Guid rootStepId)
> ##[debug]   at GitHub.Runner.Worker.JobExtension.InitializeJob(IExecutionContext jobContext, AgentJobRequestMessage message)
> ##[debug]Finishing: Set up job
> ```
> 
> How can I overcome `actions/cache` errors?

We are having a similar issue with git actions
![image](https://user-images.githubusercontent.com/70018545/194298789-ebf41b6b-3a4e-43ac-af0b-2acc4b4a0aa5.png)
